src/components/notices-provider/NoticesProvider.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import { getConfig } from '@edx/frontend-platform';
+import PropTypes from 'prop-types';
+import { getNotices } from './api';
+/**
+ * This component uses the platform-plugin-notices plugin to function.
+ * If the user has an unacknowledged notice, they will be rerouted off
+ * enterprise dashboard and onto a full-screen notice page. If the plugin is not
+ * installed, or there are no notices, we just passthrough this component.
+ */
+const NoticesProvider = ({ children }) => {
+  useEffect(() => {
+    async function getData() {
+      if (getConfig().ENABLE_NOTICES) {
+        const data = await getNotices();
+        if (data && data.results && data.results.length > 0) {
```

Comment:
nit (non-blocking): could be written as `if (data?.results?.length > 0) { ... }`